aydin/nn/models/torch/torch_unet.py

```diff
@@ -64,32 +65,41 @@ def __init__(
 
         self.conv_with_batch_norms_second_half = []
         for layer_index in range(self.nb_unet_levels):
-            if (
-                layer_index == 0
-            ):  # Handle special case input dimensions for the first layer
-                self.conv_with_batch_norms_second_half.append(
-                    ConvWithBatchNorm(
-                        self.unet_bottom_conv_out_channels,
-                        self.nb_filters
-                        * max((self.nb_unet_levels - layer_index - 2), 1),
-                        spacetime_ndim,
-                    )
-                )
+            _nb_filters = self.nb_filters * max((self.nb_unet_levels - layer_index - 2), 1)
+            # TODO: Here we assume the channel dim of input image is 1.
+            #  If we don't know the input image channel, we may want to skip the last skip-layer.
+            if self.residual:
```

Comment:
Determining the channel size is a bit tricky. This logic may break when a tricky model args or image shape was passed, but good enough to pass the test so far.

Reply:
can you think of an edge case that can break this implementation @li-li-github ?

Reply:
I don't know at this point (^^; It might break when the `nb_filters` are an odd number and the number of layers gets larger.